Re-add integration test for runtime specific components in the palette

kiersten-stokes opened the pull request:
Fixes #2298 

### What changes were proposed in this pull request?
PR #2286 removed built-in components being included by default in the palette, and the corresponding integration tests (`kfp pipeline should display custom components` and `airflow pipeline should display custom components`) were commented out. This PR creates a function to add an example component catalog for each runtime type, which allows the custom components to be displayed in the palette again.

This update also necessitates the examples packages be installed, so the `install-examples` Makefile target has been added to the existing integration test targets (`test-integration` and `test-integration-debug`).

The component id's also have been updated to reflect the latest values.

### How was this pull request tested?
Uncommented tests are now passing (locally and CI)

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

kiersten-stokes commented on Makefile:
```diff
@@ -177,10 +177,10 @@ test-ui: lint-ui test-ui-unit test-integration # Run frontend tests
 	yarn test:unit
 
-test-integration: # Run frontend cypress integration tests
+test-integration: install-examples # Run frontend cypress integration tests
 	yarn test:integration
 
-test-integration-debug: # Open cypress integration test debugger
```
Feedback wanted here: not sure if this is the best solution, but it's a relatively easy fix

marthacryan replied:
Not sure if I'm who you're looking for feedback from, but it seems fine to me? Maybe add a comment to explain why though

marthacryan reviewed (approved):
Code changes LGTM - haven't tried locally but looks like CI is working so should be good there. I would be interested to see what other frontend people think about the changes to the Makefile though (@ajbozarth @karlaspuldaro thoughts?)

karlaspuldaro reviewed (commented):
Looks good @kiersten-stokes !
I don't mind the installation command in the makefile, but maybe keeping it inside pipeline tests file could make the debug mode a bit more efficient, that way examples wouldn't be installed when testing specific files other than pipeline. 
That alternative would be removing the makefile commands and adding to `pipeline.ts` something like
  ```
before(() => {
    cy.exec('make install-examples',
      {failOnNonZeroExit: false}
    );
  });
```

ajbozarth commented:
I dont think we should be adding build targets to test targets in the Makefile. The test targets should assume that the build steps have already been run. If we need to update the CI build steps to handle this then we should do that.

I also am unsure if we should be referring to make calls from within the tests either like @karlaspuldaro suggested, those tests are also able to be run without the Makefile and should not be dependent on it.

karlaspuldaro commented:
@ajbozarth good point, what about calling the commands directly instead of depending on make like
`cy.exec('pip install --upgrade elyra-examples-kfp-catalog')`

ajbozarth commented:
@karlaspuldaro If we decided to include it in the tests then that would be how we would want to do it, but I'm still not sure we should. Don't our tests assume that the code is already build before they're run? If so, then they should assume that the examples are already installed and set up correctly. Though it is probably worth adding a check for it in the tests so they fail with a message as to why (ie that the examples aren't installed)

kiersten-stokes commented:
@karlaspuldaro and @ajbozarth great points and I agree! The integration tests start with a `make install`, though, rather than a `make install-all`, so the example components are not included and the 2 tests in question fail (and are basically obsolete anyway at that point). I like the idea of doing the `make install-all` for the integration tests instead, but I don't know where that would have to be changed/updated.

kiersten-stokes commented:
As discussed in the SCRUM, I reverted all changes to the Makefile and instead updated the CI workflow 'Install' command to include `make install-examples`.

I also added some custom error messaging to make it more obvious during troubleshooting that the error is likely with the missing example catalog connector packages.

ajbozarth commented on .github/workflows/build.yml:
```diff
@@ -142,6 +142,7 @@ jobs:
         run: |
           make yarn-install
           make only-install-server
+          make install-examples
```
in order to match our normal build order I would put this in the Setup JupyterLab section below

ajbozarth commented on tests/support/commands.ts:
```diff
@@ -59,6 +59,31 @@ Cypress.Commands.add('createRuntimeConfig', ({ type } = {}): void => {
+    cy.findByRole('menuitem', { name: /apache airflow/i }).click();
+  }
+
+  cy.findByLabelText(/^name/i).type('Example Components');
```
we should probably make this step check if one already exists first in case we end up using this in multiple tests in the future

kiersten-stokes replied:
I'm currently removing the example catalog at the very end of the `afterEach` block for the pipeline tests (similar to how test runtime configs are handled). I see what you're saying, but I wonder if we should just leave this to be dealt with when the time comes? Idk, I could go either way

Regardless, I am going to change the name of the catalog to be something like "Test Example Components" to reduce the likelihood of collisions with existing catalogs when running locally.

ajbozarth replied:
ah I see, I though that code was run once at the end, not per test. In that case ignore my comment, this way is good
